### DISCUSSION TITLE
Unable to save datetime in "YYYY-MM-DD hh:mm:ss.000Z" format

### DISCUSSION THREAD

**Discussion by rameshkumar70** [pocketbase/pocketbase#5210] [Q&A]
Pocketbase server version: 0.22.17
npm pocketbase package version: "pocketbase": "^0.21.3",

I  am using the following code on the server to obtain a form:

```
const sessionData = {
		class:  classId,
		date: form.get('date') ?? '',
		time: form.get('time') ?? '',
		duration: form.get('duration') ?? '',
		comments: form.get('comments') ?? '',
		topics: form.get('topics') ?? '',
		fee: form.get('fee') ?? 0,
	};

	 sessionData.classdatetime = `${form.get('classdatetime').replace(/T/, ' ') + '.000Z'}`;
        const rawRecord = await locals.userPb.collection('IndividualClassSession').create(sessionData);
```
The sessionData at this point prints as:

```
{
  "class": "aqfu9qsi1z4dfve",
  "date": "2024-07-12",
  "time": "11:00",
  "duration": "2",
  "comments": "",
  "topics": "",
  "fee": "0",
  "classdatetime": "2024-07-12 15:16.000Z"
}
```
The _date_ and _time_ fields are getting inserted into the collection just fine, but the _classdatetime_ field doesn't get filled up in the collection. 

![image](https://github.com/user-attachments/assets/ed50b437-aefe-4078-a085-ddb30f307830)

Looking at the debug message, I get this:

```
[10.00ms] INSERT INTO `IndividualClassSession` (`class`, `classdatetime`, `comments`, `created`, `date`, `duration`, `fee`, `id`, `time`, `topics`, `updated`) VALUES ('aqfu9qsi1z4dfve', '', '', '2024-07-12 06:12:11.399Z', '2024-07-12 00:00:00.000Z', '2', 0, 'aeuktuqcb4rp5ft', '11:00', '', '2024-07-12 06:12:11.399Z')
```

I just can't figure out why the _classdatetime_  field becomes blank. Can anyone please help? Would really appreciate it. Thanks.

**Comment by ganigeorgiev** [comment]
`classdatetime` doesn't seem to be correctly formatted (it is missing the seconds) and it is probably failing the date layout casting resulting to a zero DateTime (_you can append `:00.000Z` or just `:00` to your date_).
